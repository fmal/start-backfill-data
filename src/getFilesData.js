import fs from 'fs';
import path from 'path';

import { flattenObject, normalizePath } from './utils/index';

export default function getFilesData(filePaths) {
  return filePaths.reduce((filesData, filePath) => {
    const CWD = process.cwd();
    const pathInfo = path.parse(filePath);

    if (pathInfo.ext !== '.json') {
      return filesData;
    }

    const contents = JSON.parse(fs.readFileSync(filePath, 'utf8'));

    filesData.push({
      contents,
      flattenedContents: flattenObject(contents),
      path: normalizePath(path.relative(CWD, filePath)),
      name: pathInfo.name
    });

    return filesData;
  }, []);
}
