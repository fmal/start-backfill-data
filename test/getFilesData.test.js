import fsMock from 'fs';

import getFilesData from '../src/getFilesData';

jest.mock('fs');

describe('getFilesData', () => {
  beforeEach(() => {
    fsMock.readFileSync.mockClear();
  });

  it('extracts files data', () => {
    const FILE_PATH1 = '/path/to/file1.json';
    const FILE_PATH2 = '/path/to/file2.json';

    fsMock.__setMockFiles({
      [FILE_PATH1]: `{
        "propA": "foo",
        "propB": "bar",
        "propC": {
          "propA": "baz"
        }
      }`,
      [FILE_PATH2]: `{
        "propA": "foo",
        "propB": ["bar", "baz"]
      }`
    });

    expect(getFilesData([FILE_PATH1, FILE_PATH2])).toMatchSnapshot();
  });

  it('skips non .json files', () => {
    const FILE_PATH1 = 'file1.json';
    const FILE_PATH2 = 'file2.txt';

    fsMock.__setMockFiles({
      [FILE_PATH1]: '{}',
      [FILE_PATH2]: '{}'
    });

    getFilesData([FILE_PATH1, FILE_PATH2]);

    expect(fsMock.readFileSync).toHaveBeenCalledTimes(1);
    expect(fsMock.readFileSync).toHaveBeenCalledWith(FILE_PATH1, 'utf8');
  });
});
