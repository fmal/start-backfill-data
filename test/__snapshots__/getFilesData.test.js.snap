// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFilesData extracts files data 1`] = `
Array [
  Object {
    "contents": Object {
      "propA": "foo",
      "propB": "bar",
      "propC": Object {
        "propA": "baz",
      },
    },
    "flattenedContents": Object {
      "propA": "foo",
      "propB": "bar",
      "propC__propA": "baz",
    },
    "name": "file1",
    "path": "path/to/file1.json",
  },
  Object {
    "contents": Object {
      "propA": "foo",
      "propB": Array [
        "bar",
        "baz",
      ],
    },
    "flattenedContents": Object {
      "propA": "foo",
      "propB__0": "bar",
      "propB__1": "baz",
    },
    "name": "file2",
    "path": "path/to/file2.json",
  },
]
`;
